pallets/automation-time/src/lib.rs

```diff
@@ -1096,31 +1100,52 @@ pub mod pallet {
 			task: &TaskOf<T>,
 		) -> (Weight, Option<DispatchError>) {
 			// TODO: Handle edge case where user has enough funds to run task but not reschedule
+			if !T::DelegatorActions::is_delegation_exist(&delegator, &collator) {
+				let e = sp_runtime::DispatchErrorWithPostInfo::from(Error::<T>::DelegationDNE);
+				Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+					task_id,
+					error_message: Into::<&str>::into(e).as_bytes().to_vec(),
+					error: e,
+				});
+				return (
+					<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(),
+					Some(e.error),
+				)
+			}
+
 			let reserved_funds = account_minimum
 				.saturating_add(Self::calculate_schedule_fee_amount(&task.action, 1).expect("Can only fail for DynamicDispatch and this is always AutoCompoundDelegatedStake"));
-			match T::DelegatorActions::delegator_bond_till_minimum(
-				&delegator,
-				&collator,
-				reserved_funds,
-			) {
-				Ok(delegation) => {
-					Self::deposit_event(Event::SuccesfullyAutoCompoundedDelegatorStake {
-						task_id,
-						amount: delegation,
-					});
-					(<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(), None)
-				},
-				Err(e) => {
-					Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
-						task_id,
-						error_message: Into::<&str>::into(e).as_bytes().to_vec(),
-						error: e,
-					});
-					(
-						<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(),
-						Some(e.error),
-					)
+			match T::DelegatorActions::get_delegator_stakable_free_balance(&delegator)
+				.checked_sub(&reserved_funds)
+			{
+				Some(delegation) => {
+					match T::DelegatorActions::delegator_bond_more(
+						&delegator, &collator, delegation,
+					) {
+						Ok(_) => {
+							Self::deposit_event(Event::SuccesfullyAutoCompoundedDelegatorStake {
+								task_id,
+								amount: delegation,
+							});
+							(
+								<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(),
+								None,
+							)
+						},
+						Err(e) => {
+							Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+								task_id,
+								error_message: Into::<&str>::into(e).as_bytes().to_vec(),
+								error: e.into(),
+							});
```

Comment:
**What message will emit** when you decide not to schedule the next exeuction when there’s no more delegations?